server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
```

Comment:
Ah, I was looking at metadata resource, which returns 204, and other resources return 204. That's actually a bug in some old code I wrote :P

Thinking about things more, we should return 204. What do you think about that?